### DISCUSSION TITLE
Non-member attributes of enum should be included in completion suggestions

### DISCUSSION THREAD

**Discussion by lambda674** [microsoft/pylance-release#5744] [Enhancement]
It's possible to define a non member attribute that is recognized with Pyright:
```python
from enum import Enum

class MyEnum(Enum):
    other_val: int

    def __new__(cls, name: str, other_val: int) -> "MyEnum":
        obj = str.__new__(cls, name)
        obj._value_ = name

        obj.other_val = other_val
        return obj

    ONE = ("One", 1)
    TWO = ("Two", 2)

print(MyEnum.ONE.other_val) # No error
print(MyEnum.ONE.some_random_val) # Cannot access member "some_random_val" for type "Literal[MyEnum.One]" Member "random-val" is unknown
```

But when I use Pyright LSP in Neovim, I don't see `other_val` as a property under `MyEnum.ONE` (let me know if there's a way to show an example or describe this better). The only `Property` I see are `name`, `value`, and `__class__`. The reason this matters is that I don't get autocomplete for non member attributes, even though Pyright is aware of the attribute.

I've tried using a stub file without success (including trying to use a property function):
```python
from enum import Enum

class MyEnum(Enum):
    ONE: str
    TWO: str
    @property
    def other_val(self) -> int: ...
```

[I've seen the recommendation that using a property would probably be better](https://github.com/microsoft/pyright/discussions/4883#discussioncomment-5494983) so I tried that, but still no success (I would prefer the first implementation but this one is fine):
```python
from enum import Enum

class MyEnum(Enum):
    ONE = "one"
    TWO = "two"

    @property
    def other_val(self) -> int:
        vals = {
            MyEnum.ONE: 1,
            MyEnum.TWO: 2,
        }
        return vals[self]
```

#### Versions
pyright: 1.1.357
Python: 3.10.13

**Comment by erictraut** [comment]
Changing this from a bug to an enhancement request.

Enums require a bunch of special-case handling. This use case is pretty esoteric, so it's not clear whether it's worth the effort to add this special-case logic to the completion provider. I'll let the pylance team decide whether they want to invest in this.

Could someone from the pylance team please transfer this feature request to the pylance-release project? @rchiodo, @debonte

**Comment by debonte** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.
